client/devdocs/controller.js

```diff
@@ -132,14 +133,17 @@ const devdocs = {
 	},
 
 	pleaseLogIn: function( context ) {
+		const currentUrl = url.parse( location.href );
+		const redirectUrl = currentUrl.protocol + '//' + currentUrl.host + '/devdocs/welcome';
+
 		ReactDom.unmountComponentAtNode( document.getElementById( 'secondary' ) );
 
 		ReactDom.render(
 			React.createElement( EmptyContent, {
 				title: 'Log In to start hacking',
 				line: 'Required to access the WordPress.com API',
 				action: 'Log In to WordPress.com',
-				actionURL: 'https://wordpress.com/wp-login.php?redirect_to=http%3A%2F%2Fcalypso.localhost%3A3000/devdocs/welcome',
+				actionURL: 'https://wordpress.com/wp-login.php?redirect_to=' + encodeURIComponent( redirectUrl ),
```

Comment:
I'd prefer not to use that here, as it's a shorter line without the interpolation :)